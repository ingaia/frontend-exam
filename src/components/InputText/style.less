@import './../../variables.less';

.input-group {
    position: relative;
    height: 40px;
    width: 100%;
    margin-bottom: 15px;

    input {
        height: 100%;
        width: 100%;
        padding-top: 10px;
        background: transparent;
        border: none;
        color: #FFFFFF;
        font-size: 14px;
        outline: none;

        &::placeholder {
            color: transparent;
        }

        &:focus + .label .label-content,
        &:not(:placeholder-shown)+ .label .label-content {
            transform: translateY(-150%);  
            font-size: 11px; 
        }
    }
    
    label {
        position: absolute;
        bottom: 0px;
        left: 0%;
        height: 100%;
        width: 100%;
        border-bottom: 1px solid @hillary;
        color: @hillary;
        pointer-events: none;

        &:after {
            content: '';
            position: absolute;
        }
    }

    .label-content {
        position: absolute;
        bottom: 5px;
        left: 0px;
        transition: all 0.3s ease;
    }
}