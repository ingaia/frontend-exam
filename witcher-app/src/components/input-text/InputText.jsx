import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import styles from './InputText.module.scss';

const InputText = (props) => {

    const { id, type, placeholder, value, onChange, className } = props

    return (
        <input 
            id={id}
            type={type}
            placeholder={placeholder}
            value={value}
            onChange={onChange}
            className={classnames(styles.InputText, className)}
        />
    );
}
 
InputText.propTypes = {
    id: PropTypes.string.isRequired,
    type: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string,
}

InputText.defaultProps = {
    type: 'text',
    className: '',
}

export default InputText;