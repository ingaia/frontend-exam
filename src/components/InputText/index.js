import React from 'react';

import './style.less';

export default function InputText(props) {
    return (
        <div className="input-group">
            <input type={props.type ? props.type : 'text'} 
                name={props.name}
                value={props.value}
                onChange={props.onChange}
                placeholder={props.label}
                required={props.required}/>
            <label className="label" htmlFor={props.name}>
                <span className="label-content">
                    {props.label}
                </span>
            </label>
        </div>
    )
}