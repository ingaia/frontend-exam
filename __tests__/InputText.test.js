import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import InputText from '../src/components/InputText';

describe('<InputText/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<InputText />);
    });

    it('should render', () => {
        wrapper.setProps({
            label: 'Email',
            name: 'email'
        });

        const input = wrapper.children('input');
        const label = wrapper.children('label');

        expect(label.text()).toBe('Email');
        expect(input.exists('[htmlFor="email"]'))
        expect(input.exists('[type="text"]')).toBe(true);
        expect(input.exists('[name="email"]')).toBe(true);
        expect(wrapper.hasClass('input-group')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should the input a type email', () => {
        wrapper.setProps({type: 'email'});
        const input = wrapper.children('input');
        expect(input.exists('[type="email"]')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});