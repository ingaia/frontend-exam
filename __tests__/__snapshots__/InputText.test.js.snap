// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText/> should render 1`] = `
<div
  className="input-group"
>
  <input
    name="email"
    placeholder="Email"
    type="text"
  />
  <label
    className="label"
    htmlFor="email"
  >
    <span
      className="label-content"
    >
      Email
    </span>
  </label>
</div>
`;

exports[`<InputText/> should the input a type email 1`] = `
<div
  className="input-group"
>
  <input
    type="email"
  />
  <label
    className="label"
  >
    <span
      className="label-content"
    />
  </label>
</div>
`;
